src/en/space-station-14/chat.md

```diff
@@ -0,0 +1,495 @@
+
+| Commuication type				| Details  	|
+| :---:							| ---		|
+| Hivemind  					| Blobs and other oddities usually can talk using some sort of bio-radio - hivemind, which is like real radio, but - messages are not being forwarded to whisper on talking, and when hivemind core members are dead - it should collapse (at least chat is not available). And there is no reason to have same hive-mind radio if there is more then one main blob, for example - such chat should have id of network and round should be able to have more then 1 of such chat at the same time, even for same antag type. As for hivemind core - blob have single core, but for example changeling could have multiple main ones. |
+| Telepathy  					| vamps and brain-parasites could communicate using direct messaging, selecting one target and talking 'as kinda hivemind between 2 targets' but only in one direction. |
+| PDA DM						| PDA needs messaging feature which is ofc PDA app and depends on separate system, but it also requires to leave notification for player in chat - that they received message on PDA. Maybe text of message should be included. It is sort of direct messaging. |
+
+## Functional requirements
+Now what we looked at in previous parts, lets try to aggregate it into unique requirements.
+
+From main standpoint of chat we can deduce that most of requirements will be concentrated around two processes - **emitting (producing) message**, and **consuming** (and rendering, as a result). **We will call process in between oh those 'chat pipeline'.** But sometimes that process is more complex - consuming device could be just relay that transfers and emits almost the same message by itself - some kind of **forwarding**. *Example of such thing is portable radio, which is listening and broadcasting what it hears.*
+
+We can state that there is always 1 producer, and 1 or more consumers of messages. Both sides can be either player or some in-game entity. Chat pipeline exact process can be different, based on aspects of entity, environment, and way of communication. Consumers could be not only players, but **also devices**. This is important part as it brings extensibility to next level - all sort of devices becomes available when routing of messages can be done through multiple layers of devices without too much of hand-crafted c# logic.
+
+Nature of ss14 development is fluent and there is a lot of demand for features to be customizable within yaml prototypes, without having 'fun' of writing and testing c# code. It is better to create new chat system with this in mind. As previously said, we have producing and consuming, so lets try to find out what configuration/configurability they need.
+
+From **message producing** standpoint
+
+| Condition				| Technical description | Description 			|
+|:---					|:---			|:---				|
+|Is cautious			| Does not have StunnedComponent, MutedComponent or GhostComponent |Can only speak if alive and awake |
```

Comment:
I assume you mean conscious, and not cautious here?

Reply:
fixed